File: apps/backend/src/EnterpriseAgentOs.Domain/Interfaces/AgentLogs/IAgentLogRepository.cs
namespace EnterpriseAgentOs.Domain.Interfaces.AgentLogs;

public sealed record GlobalLogRow(EnterpriseAgentOs.Domain.Models.AgentLogRecord Log, string AgentName);

public interface IAgentLogRepository
{
    Task<List<EnterpriseAgentOs.Domain.Models.AgentLogRecord>> ListAsync(Guid agentId, DateTime? before, int limit, CancellationToken ct = default);
    Task<(List<GlobalLogRow> Items, int Total)> ListGlobalAsync(
        string? search, string? agentName, EnterpriseAgentOs.Domain.Models.AgentLogType? type, int skip, int limit, CancellationToken ct = default);
    Task<EnterpriseAgentOs.Domain.Models.AgentLogRecord> AppendAsync(EnterpriseAgentOs.Domain.Models.AgentLogRecord record, CancellationToken ct = default);
    Task AppendPairAsync(EnterpriseAgentOs.Domain.Models.AgentLogRecord toolCall, EnterpriseAgentOs.Domain.Models.AgentLogRecord toolResult, CancellationToken ct = default);
    Task<EnterpriseAgentOs.Domain.Models.AgentLogRecord?> GetByIdAsync(Guid id, CancellationToken ct = default);
    Task<(List<EnterpriseAgentOs.Domain.Models.AgentLogRecord> Items, int Total)> GetToolCallsAsync(Guid agentId, int limit, int offset, CancellationToken ct = default);
    Task<Dictionary<string, EnterpriseAgentOs.Domain.Models.AgentLogRecord>> GetResultsByCorrelationAsync(Guid agentId, IReadOnlyCollection<string> correlationIds, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/analytics/IAgentLogRepository.java
package com.enterpriseagentos.backendjava.domain.features.analytics;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface IAgentLogRepository {
    CompletableFuture<List<AgentLogRecord>> listAsync(UUID agentId, Instant before, int limit);
    CompletableFuture<List<AgentLogRecord>> listAfterAsync(UUID agentId, UUID afterLogId, int limit);
    CompletableFuture<AgentLogRecord> appendAsync(AgentLogRecord record);
    CompletableFuture<Void> appendPairAsync(AgentLogRecord toolCall, AgentLogRecord toolResult);
    CompletableFuture<AgentLogRecord> getByIdAsync(UUID id);
    CompletableFuture<Map<String, AgentLogRecord>> getResultsByCorrelationAsync(UUID agentId, Collection<String> correlationIds);
    CompletableFuture<Void> deleteByAgentIdsAsync(List<UUID> agentIds);
    CompletableFuture<List<AgentLogRecord>> listByAgentIdsAsync(List<UUID> agentIds, List<AgentLogType> types);
}
